examples/nft-nd/nft.go

```diff
@@ -285,3 +287,59 @@ func Properties(id []byte) map[string]string {
 	}
 	return result
 }
+
+// RoyaltyRecipient contains information about the recipient and the royalty amount.
+type RoyaltyRecipient struct {
+	Address interop.Hash160
+	Amount  int
+}
+
+// RoyaltyInfo returns a list of royalty recipients and the corresponding royalty amounts.
+func RoyaltyInfo(tokenID []byte, royaltyToken interop.Hash160, salePrice int) []RoyaltyRecipient {
+	if salePrice <= 0 {
+		panic("sale price must be positive")
+	}
+
+	executingHash := runtime.GetExecutingScriptHash()
+	if !royaltyToken.Equals(executingHash) {
+		panic("invalid royalty token")
+	}
+
+	ctx := storage.GetReadOnlyContext()
+	owner := getOwnerOf(ctx, tokenID)
+	if owner == nil {
+		panic("invalid token ID")
+	}
+
+	royaltyInfoKey := append([]byte(royaltyInfoPrefix), tokenID...)
+	val := storage.Get(ctx, royaltyInfoKey)
+
+	var recipients []RoyaltyRecipient
+	if val == nil {
+		recipients = []RoyaltyRecipient{
+			{
+				Address: owner,
+				Amount:  salePrice / 10,
+			},
+		}
+	} else {
+		storedRecipients := val.([]RoyaltyRecipient)
+		for _, r := range storedRecipients {
+			recipients = append(recipients, RoyaltyRecipient{
+				Address: r.Address,
+				Amount:  salePrice / r.Amount,
+			})
+		}
+	}
+	return recipients
+}
+
+// SetRoyaltyInfo sets the royalty share for specified recipients on a given token ID.
+// Only the token owner can set the royalty information.
+func SetRoyaltyInfo(ctx storage.Context, tokenID []byte, recipients []RoyaltyRecipient) bool {
```

Comment:
>  recipients []RoyaltyRecipient

Don't use RoyaltyRecipient for this, create a separate structure, because RoyaltyRecipient has Amount field, and SetRoyaltyInfo needs a *share* instead of absolute amount value.